test/SecretManager.Tests/ConfigurationExtensionTests.cs

```diff
@@ -28,13 +28,7 @@ public void AddUserSecrets_Does_Not_Fail_On_Non_Existing_File()
             var backupServiceProvider = CallContextServiceLocator.Locator.ServiceProvider;
```

Comment:
You can remove backing up the `ServiceProvider` and also remove the finally block.